Description: update the makefile Change CFLAGS and LDFLAGS and change the binary name generated by posixXOP
Author: Sophie Brun <sophie@freexian.com>
Origin: vendor
Last-Update: 2016-01-11
--- a/src/Makefile
+++ b/src/Makefile
@@ -32,9 +32,9 @@ endif
 
 # -mstackrealign is required otherwise crashes on windows xp, see hashcat v0.41 and v0.42
 
-CFLAGS  := -W -Wall -pipe -I include/ -O2 -s -fomit-frame-pointer -funroll-loops
+CFLAGS  += -W -Wall -pipe -I include/ -O2 -s -fomit-frame-pointer -funroll-loops
 #CFLAGS  := -W -Wall -pipe -I include/ -g
-LDFLAGS :=
+LDFLAGS ?=
 
 all: binaries
 
@@ -110,7 +110,7 @@ $(DIR_OSX64)/common.OSX.64.o: src/common
 ##
 
 DIR_POSIX32      = obj
-LIBGMP_POSIX32   = /opt/hashcat-deps/gmp/linux32
+#LIBGMP_POSIX32   = /opt/hashcat-deps/gmp/linux32
 CC_POSIX32       = gcc
 CFLAGS_POSIX32   = $(CFLAGS) -I$(LIBGMP_POSIX32)/include -D__HC_x86_32__ -DPOSIX -m32 -msse2
 LDFLAGS_POSIX32  = $(LDFLAGS) -L$(LIBGMP_POSIX32)/lib -lm -lpthread -lgmp
@@ -125,7 +125,7 @@ rules-debug32.bin: $(DIR_POSIX32)/rp.LIN
 hashcat-posix32: hashcat-cli32.bin
 
 hashcat-cli32.bin: $(DIR_POSIX32)/tsearch.LINUX.32.o $(DIR_POSIX32)/rp.LINUX.32.o  $(DIR_POSIX32)/engine.LINUX.32.o src/hashcat-cli.c
-	$(CC_POSIX32) $(CFLAGS_POSIX32) $(DIR_POSIX32)/*.LINUX.32.o src/hashcat-cli.c -o hashcat-cli32.bin $(LDFLAGS_POSIX32)
+	$(CC_POSIX32) $(CFLAGS_POSIX32) $(CPPFLAGS) $(DIR_POSIX32)/*.LINUX.32.o src/hashcat-cli.c -o hashcat $(LDFLAGS_POSIX32)
 
 engine-posix32: $(DIR_POSIX32)/engine.LINUX.32.o
 
@@ -152,7 +152,7 @@ $(DIR_POSIX32)/common.LINUX.32.o: src/co
 ##
 
 DIR_POSIX64      = obj
-LIBGMP_POSIX64   = /opt/hashcat-deps/gmp/linux64
+#LIBGMP_POSIX64   = /opt/hashcat-deps/gmp/linux64
 CC_POSIX64       = gcc
 CFLAGS_POSIX64   = $(CFLAGS) -I$(LIBGMP_POSIX64)/include -D__HC_x86_64__ -DPOSIX -m64 -msse2
 LDFLAGS_POSIX64  = $(LDFLAGS) -L$(LIBGMP_POSIX64)/lib -lm -lpthread -lgmp
@@ -167,7 +167,7 @@ rules-debug64.bin: $(DIR_POSIX64)/rp.LIN
 hashcat-posix64: hashcat-cli64.bin
 
 hashcat-cli64.bin: $(DIR_POSIX64)/tsearch.LINUX.64.o $(DIR_POSIX64)/rp.LINUX.64.o $(DIR_POSIX64)/engine.LINUX.64.o src/hashcat-cli.c
-	$(CC_POSIX64) $(CFLAGS_POSIX64) $(DIR_POSIX64)/*.LINUX.64.o src/hashcat-cli.c -o hashcat-cli64.bin $(LDFLAGS_POSIX64)
+	$(CC_POSIX64) $(CFLAGS_POSIX64) $(CPPFLAGS) $(DIR_POSIX64)/*.LINUX.64.o src/hashcat-cli.c -o hashcat $(LDFLAGS_POSIX64)
 
 engine-posix64: $(DIR_POSIX64)/engine.LINUX.64.o
 
@@ -283,17 +283,17 @@ CC_POSIXXOP       = gcc
 CFLAGS_POSIXXOP   = $(CFLAGS) -I$(LIBGMP_POSIXXOP)/include -D__HC_XOP__ -DPOSIX -m64 -mxop
 LDFLAGS_POSIXXOP  = $(LDFLAGS) -L$(LIBGMP_POSIXXOP)/lib -lm -lpthread -lgmp
 
-posixXOP: hashcat-cliXOP.bin
+posixXOP: hashcat-cliXOP
 
 rules-posixXOP: rules-debugXOP.bin
 
 rules-debugXOP.bin: $(DIR_POSIXXOP)/rp.LINUX.XOP.o src/rules-debug.c
 	$(CC_POSIXXOP) $(CFLAGS_POSIXXOP) $(DIR_POSIXXOP)/*.LINUX.XOP.o src/rules-debug.c -o rules-debugXOP.bin $(LDFLAGS_POSIXXOP)
 
-hashcat-posixXOP: hashcat-cliXOP.bin
+hashcat-posixXOP: hashcat-cliXOP
 
-hashcat-cliXOP.bin: $(DIR_POSIXXOP)/tsearch.LINUX.XOP.o $(DIR_POSIXXOP)/rp.LINUX.XOP.o $(DIR_POSIXXOP)/engine.LINUX.XOP.o src/hashcat-cli.c
-	$(CC_POSIXXOP) $(CFLAGS_POSIXXOP) $(DIR_POSIXXOP)/*.LINUX.XOP.o src/hashcat-cli.c -o hashcat-cliXOP.bin $(LDFLAGS_POSIXXOP)
+hashcat-cliXOP: $(DIR_POSIXXOP)/tsearch.LINUX.XOP.o $(DIR_POSIXXOP)/rp.LINUX.XOP.o $(DIR_POSIXXOP)/engine.LINUX.XOP.o src/hashcat-cli.c
+	$(CC_POSIXXOP) $(CFLAGS_POSIXXOP) $(CPPFLAGS) $(DIR_POSIXXOP)/*.LINUX.XOP.o src/hashcat-cli.c -o hashcat-xop $(LDFLAGS_POSIXXOP)
 
 engine-posixXOP: $(DIR_POSIXXOP)/engine.LINUX.XOP.o
 
